### PR TITLE
Add latency perf test

### PR DESCRIPTION
**Pull request by letmaik**
This implements some of #50, in particular the latency for `POST /entries` as well as time-to-receipt which involves further GET calls. It is implemented as a pytest that's disabled by default. It writes a `perf.json` to the current folder (added to `.gitignore`) which gets uploaded as pipeline artifact in CI.

### CODE REVIEW

File: .pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ stages:
           Pool: 1es-dcv3
           Environment:
             ENABLE_PREFIX_TREE: OFF
+          FunctionalTestArguments: '--enable-perf'
```
Comment (letmaik) [inline .pipelines/azure-pipelines.yml:41]: If this slows down CI too much we may also introduce a separate pipeline that runs on commits to `main`, not in PRs.

File: pyscitt/pyscitt/client.py [hunk 2]
```diff
@@ -153,6 +153,7 @@ def request(
         url,
         *,
         retry_on=[],
+        wait_time=None,
```
Comment (plietar) [inline pyscitt/pyscitt/client.py:156]: What about making this a property of the Client instead? You can easily create a "modified" client in tests by taking the `client` fixture and calling `client.replace(wait_time=0)`.

See the authentication tests, where it gets used to set credentials: https://github.com/microsoft/scitt-ccf-ledger/blob/566de22714d6ebc7745a8582a617da2ef5d832a8/test/test_auth.py#L50

File: test/test_perf.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        # the CCF network and is hence just a rough estimate.
+        stats = {
+            "latency_did_web_submit_s": latency_did_web_submit_s,
+            "latency_did_web_submit_and_receipt_s": latency_did_web_submit_and_receipt_s,
+            "latency_x5c_submit_s": latency_x5c_submit_s,
+            "latency_x5c_submit_and_receipt_s": latency_x5c_submit_and_receipt_s,
+        }
+
+        # Write stats to file for further analysis.
+        with open("perf.json", "w", encoding="utf-8") as fp:
```
Comment (plietar) [inline test/test_perf.py:82]: This is probably fine for now, but eventually I think we'll want some pattern for reporting metrics from multiple different tests.

File: .pipelines/common.yml [hunk 4]
```diff
@@ -77,3 +80,9 @@ jobs:
 
       - script: ./run_functional_tests.sh ${{ parameters.FunctionalTestArguments }}
         displayName: Run Functional Tests
+
+      - ${{ if parameters.PublishPerfResults }}:
+        - task: PublishPipelineArtifact@1
+          inputs:
+            artifactName: "perf-results"
```
Comment (plietar) [inline .pipelines/common.yml:87]: Looking at the DevOps page, it seems all the jobs in the pipeline are writing to the same artifact name and overwriting each other. I think the `perf-results` name is going to need to be parametrized.

Maybe this works?
```suggestion
            artifactName: "${{ parameters.Name }}-perf-results"
```

**Review by plietar — CHANGES_REQUESTED** [review]
This is probably going to suffer from the Nagle / delayed ACK issue I've seen before. The proper way to fix it would probably be to make a feature request on httpx.

File: .pipelines/common.yml (hunk 4 above)
Comment (letmaik) [reply .pipelines/common.yml:87]: Yeah, for now there's only a single job that runs the perf test, but you're right, we'll probably extend that to more jobs in the future.

Comment (plietar) [reply .pipelines/common.yml:87]: Oh I hadn't even realized that. I wonder what happens if multiple jobs write the same artifact then? Does it throw an error?

Comment (letmaik) [reply .pipelines/common.yml]: Yes, you can't override.

**Review by plietar — APPROVED** [review]

**Pull request merged**